src/rust-crypto/backup.ts

```diff
@@ -164,9 +165,10 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         // We force a check to ensure to have the latest version. We also want to check that the backup is trusted
         // as we don't want to store the secret if the backup is not trusted, and eventually import megolm keys later from an untrusted backup.
         const backupCheck = await this.checkKeyBackupAndEnable(true);
+        const version = backupCheck?.backupInfo.version;
 
-        if (!backupCheck?.backupInfo?.version || !backupCheck.trustInfo.trusted) {
-            // There is no server-side key backup, or the backup is not signed by a trusted cross-signing key or trusted own device.
+        if (!backupCheck || !version) {
+            // There is no server-side key backup, or the backup is not trusted.
```

Comment:
So is this one

```suggestion
            // There is no server-side key backup.
```

Reply:
I think current comment make more sense here, as we want to early return even when server-side backup exist but program failed to trust it.
Should we just remove this comment as `checkKeyBackupAndEnable` docs explain that `KeyBackupCheck` will be return only when backup exist and is trusted?